### ISSUE TITLE
Connection with providers not working

### ISSUE THREAD

**Issue by davidmarcoes** [strapi/strapi#1491]
**Informations**
- **Node.js version**: 9.8.0
- **npm version**: 5.6.0
- **Strapi version**: 3.0.0-alpha.12.6
- **Database**: mongodb
- **Operating system**: macOS High Sierra

**What is the current behavior?**
Different values on Auth.js#providers `grantConfig` variable depending on some provider being enabled/disabled/modified.

Depending on how many of the previous operations are executed, sometimes the expected value is setted:
```Object {email: Object, facebook: Object, google: Object, github: Object, twitter: Object, …}
email:Object {enabled: true, icon: "envelope"}
facebook:Object {enabled: false, icon: "facebook-official", key: "", …}
github:Object {enabled: false, icon: "github", key: "", …}
google:Object {enabled: false, icon: "google", key: "", …}
server:Object {protocol: "http", host: "localhost:1337"}
twitter:Object {enabled: false, icon: "twitter", key: "", …}
__proto__:Object {constructor: , __defineGetter__: , __defineSetter__: , …}
```

While some others the parent of the expected variable is setted instead:
```
Object {providers: Object, server: Object}
providers:Object {email: Object, facebook: Object, google: Object, …}
server:Object {protocol: "http", host: "localhost:1337"}
__proto__:Object {constructor: , __defineGetter__: , __defineSetter__: , …}
```

Additionally and probably because of this, under some situations enabled providers become disabled and the config is lost after a failure.

**Steps to reproduce the problem**
- Enable a provider on the admin panel

- Call /connect/:provider

**What is the expected behavior?**
/connect/:provider is able to communicate and authenticate some user

**Suggested solutions**
Mappings against the store

**Comment by lauriejim** [comment]
Hello @davidmarcoes did you follow https://strapi.io/documentation/guides/authentication.html#providers documentation ?

I'm not sure to 100% understand the issue.

**Comment by daqst** [comment]
I think this issue is addressed and solved in issue #1512 

The problem is when on the providers tab in roles & permissions, changing provider settings and clicking save does not send just the 'providers' object, rather an object with 'providers' as one of the keys.  This means that the code looking for provider settings (user-permissions/controllers/Auth.js:150) looks in the wrong place.

**Comment by davidmarcoes** [comment]
Issued in #1512
